source/custom/visa_service.custom.cpp

```diff
@@ -458,21 +457,23 @@ ::grpc::Status VisaService::UsbControlIn(::grpc::ServerContext* context, const U
     ViUInt16 w_value = request->w_value();
     ViUInt16 w_index = request->w_index();
     ViUInt16 w_length = request->w_length();
-    std::string buffer(return_count, '\0');
+    std::vector<ViByte> buffer(w_length);
     ViUInt16 return_count{};
-    auto status = library_->UsbControlIn(vi, bm_request_type, b_request, w_value, w_index, w_length, (ViByte*)buffer.data(), &return_count);
-    if (!status_ok(status)) {
+    auto status = library_->UsbControlIn(vi, bm_request_type, b_request, w_value, w_index, w_length, buffer.data(), &return_count);
+    if (!status_ok(status) && return_count == 0) {
       return ConvertApiErrorStatusForViSession(context, status, vi);
     }
     response->set_status(status);
-    response->set_buffer(buffer);
+    response->set_buffer(buffer.data(), return_count);
     response->set_return_count(return_count);
     return ::grpc::Status::OK;
   }
+  catch (std::bad_alloc) {
```

Comment:
We should catch by const reference:
`catch (const std::bad_alloc&) {`
Please fix the other catch blocks in this file as well.

Reply:
Commit 2.